PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift

```diff
@@ -2,101 +2,105 @@
 //  DiningVenueDetailMenuView.swift
 //  PennMobile
 //
-//  Created by CHOI Jongmin on 23/6/2020.
-//  Copyright © 2020 PennLabs. All rights reserved.
+//  Created by Jon Melitski on 2/26/2024.
+//  Copyright © 2024 PennLabs. All rights reserved.
 //
 
 import SwiftUI
 import WebKit
 import PennMobileShared
 
 struct DiningVenueDetailMenuView: View {
-    var menus: [DiningMenu]
+    
+    @EnvironmentObject var diningVM: DiningViewModelSwiftUI
+    
     var id: Int
-    var venue: DiningVenue?
+    var venue: DiningVenue
+    var parentScrollProxy: ScrollViewProxy
+    
+    /// Notable invariant, the menuDate must ALWAYS match all of the menus in the array.
     @State var menuDate: Date
-    @State private var menuIndex: Int
-    @State private var showMenu: Bool
-    @EnvironmentObject var diningVM: DiningViewModelSwiftUI
-    init(menus: [DiningMenu], id: Int, venue: DiningVenue? = nil, menuDate: Date = Date(), showMenu: Bool = false) {
-        self.menus = menus
+    @State var menus: [DiningMenu]
+    
+    // Both are nil on init
+    @State private var currentMenu: DiningMenu?
+    @State private var selectedStation: DiningStation?
+    
+    @Binding private var parentScrollOffset: CGPoint
+    
+    init(menus: [DiningMenu], id: Int, venue: DiningVenue, menuDate: Date = Date(), parentScrollProxy: ScrollViewProxy, parentScrollOffset: Binding<CGPoint>) {
         self.id = id
         self.venue = venue
-        _showMenu = State(initialValue: showMenu)
+        self.parentScrollProxy = parentScrollProxy
+        _parentScrollOffset = parentScrollOffset
+        _menus = State(initialValue: menus)
         _menuDate = State(initialValue: menuDate)
-        _menuIndex = State(initialValue: 0)
-        _menuIndex = State(initialValue: self.getIndex())
+        _currentMenu = State(initialValue: getMenu())
+        _selectedStation = State(initialValue: currentMenu?.stations.first ?? nil)
+        
     }
-    func getIndex() -> Int {
-        var inx = 0
-        if self.venue != nil && Calendar.current.isDate(self.menuDate, inSameDayAs: Date()) {
-            if let meal = self.venue!.currentOrNearestMeal {
-                inx = self.menus.firstIndex { $0.service == meal.label } ?? inx
-            }
+    
+    /// Constraints of this function:
+    /// Need to know if a meal is currently going on, to return it.
+    /// If there is a meal today that is closest (utilities), return it.
+    /// If the selected date is not the current day, return the first menu.
+    /// If at any point, the list of menus is empty, return nil.
+    func getMenu() -> DiningMenu? {
+        if (menus.count == 0) { return nil }
+        
+        if (!Calendar.current.isDate(menuDate, inSameDayAs: Date())) {
```

Comment:
We should probably make this timezone-aware at some point (namely forcing the timezone to always be `America/New_York`) but we can deal with that in another PR (ideally we'd add an extension to `Calendar`)